connectors/src/connectors/notion/lib/notion_api.ts

```diff
@@ -374,15 +382,26 @@ async function getBlockParent(
           })(blockParent);
           return null;
       }
+
+      depth += 1;
+      if (depth === max_depth) {
+        // We don't want to go up more than 8 levels.
+        return null;
+      }
     } catch (e) {
```

Comment:
If we have a series of errors but they are not isAPIResponseError then we have an infinite loop no?

Reply:
Oups yes

Reply:
Yep

Reply:
Fixed